server/filestore.go

```diff
@@ -1770,7 +1777,7 @@ func (mb *msgBlock) flushPendingMsgs() error {
 
 	// Only one can be flushing at a time.
 	mb.setFlushing()
-	defer mb.clearFlushing()
+	defer mb.clearFlushingUnlocked()
```

Comment:
Would it not be worth moving the original "defer mb.clearFlushing()" after line 1811 and call clearFlushingUnlocked() after write error before returning (around line 1795).

Arguably, defer could even be omitted because after the write and when reacquiring the lock there is currently only 2 places were we return, so could call in place.

The current approach seem just a bit weird because we release the lock and at the execution of this defer will reacquire the lock just to clear the "flag".

Reply:
Yes feels weird(ish) but wanted the defer to make sure it gets done and we do not return and forget..

Reply:
However let me take another look.

Reply:
ok take a look now.